### PR TITLE
house: transform output to array of whole sentence

### LINKED ISSUE #1200: house: Inconsistent expected output with other recitation-related exercises
Currently there are four exercises related with reciting verses: [beer-song](https://github.com/exercism/problem-specifications/blob/master/exercises/beer-song/canonical-data.json), [food-chain](https://github.com/exercism/problem-specifications/blob/master/exercises/food-chain/canonical-data.json), [house](https://github.com/exercism/problem-specifications/blob/master/exercises/house/canonical-data.json), and [twelve-days](https://github.com/exercism/problem-specifications/blob/master/exercises/twelve-days/canonical-data.json).

## Testing a single verse

In **beer-song**:

```
"expected": [
  "99 bottles of beer on the wall, 99 bottles of beer.",
  "Take one down and pass it around, 98 bottles of beer on the wall."
]
```

In **food-chain**:

```
"expected": [
  "I know an old lady who swallowed a spider.",
  "It wriggled and jiggled and tickled inside her.",
  "She swallowed the spider to catch the fly.",
  "I don't know why she swallowed the fly. Perhaps she'll die."
]
```

In **twelve-days**:

```
"expected": [
  "On the first day of Christmas my true love gave to me, a Partridge in a Pear Tree."
]
```

But in **house**, a single sentence is separated into multiple strings:

```
"expected": [
  "This is the dog",
  "that worried the cat",
  "that killed the rat",
  "that ate the malt",
  "that lay in the house that Jack built."
]
```

## Testing multiple verses

In **beer-song**, verses are separated by an empty string:

```
"expected": [
  "2 bottles of beer on the wall, 2 bottles of beer.",
  "Take one down and pass it around, 1 bottle of beer on the wall.",
  "",
  "1 bottle of beer on the wall, 1 bottle of beer.",
  "Take it down and pass it around, no more bottles of beer on the wall.",
  "",
  "No more bottles of beer on the wall, no more bottles of beer.",
  "Go to the store and buy some more, 99 bottles of beer on the wall."
]
```

And it's the same case in **food-chain**:

```
"expected": [
  "I know an old lady who swallowed a fly.",
  "I don't know why she swallowed the fly. Perhaps she'll die.",
  "",
  "I know an old lady who swallowed a spider.",
  "It wriggled and jiggled and tickled inside her.",
  "She swallowed the spider to catch the fly.",
  "I don't know why she swallowed the fly. Perhaps she'll die.",
  "",
  "I know an old lady who swallowed a bird.",
  "How absurd to swallow a bird!",
  "She swallowed the bird to catch the spider that wriggled and jiggled and tickled inside her.",
  "She swallowed the spider to catch the fly.",
  "I don't know why she swallowed the fly. Perhaps she'll die."
]
```

So is **house**:

```
"expected": [
  "This is the cat",
  "that killed the rat",
  "that ate the malt",
  "that lay in the house that Jack built.",
  "",
  "This is the dog",
  "that worried the cat",
  "that killed the rat",
  "that ate the malt",
  "that lay in the house that Jack built.",
  "",
  "This is the cow with the crumpled horn",
  "that tossed the dog",
  "that worried the cat",
  "that killed the rat",
  "that ate the malt",
  "that lay in the house that Jack built."
]
```

But in **twelve-days**, there are no separating empty strings:

```
"expected": [
  "On the fourth day of Christmas my true love gave to me, four Calling Birds, three French Hens, two Turtle Doves, and a Partridge in a Pear Tree.",
  "On the fifth day of Christmas my true love gave to me, five Gold Rings, four Calling Birds, three French Hens, two Turtle Doves, and a Partridge in a Pear Tree.",
  "On the sixth day of Christmas my true love gave to me, six Geese-a-Laying, five Gold Rings, four Calling Birds, three French Hens, two Turtle Doves, and a Partridge in a Pear Tree."
]
```

## My suggestion

Since each verse in both **house** and **twelve-days** is a single long sentence (contrarily **beer-song** and **food-chain** have multiple sentences in their verses), I would suggest transforming **house**'s output to the same format as **twelve-day**'s, because IMO separating a single sentence into multiple pieces is not a natural choice.

### PR DESCRIPTION
**Pull request by mRcfps**
Closes #1200.

* Bump version
* Remove redundant reasons for separating a verse into strings
* Transform the output to the same format as twelve-days

### CODE REVIEW

File: exercises/house/canonical-data.json [hunk 1]
```diff
@@ -1,10 +1,7 @@
 {
   "exercise": "house",
-  "version": "2.1.0",
+  "version": "2.2.0",
   "comments": [
-    "JSON doesn't allow for multi-line strings, so all verses are presented ",
```
Comment (ErikSchierboom) [inline exercises/house/canonical-data.json]: Any reason to remove this comment? As far as I can tell, it still holds true when multiple verses are recited. The [food-chain](https://github.com/exercism/problem-specifications/blob/master/exercises/food-chain/canonical-data.json#L5) canonical data also has this comment.

**Comment by petertseng** [comment]
I have determined that under the following diff to the verifier, this JSON file verifies.

```diff
diff --git i/exercises/house/verify.rb w/exercises/house/verify.rb
index ea4456a1..c013f33d 100644
--- i/exercises/house/verify.rb
+++ w/exercises/house/verify.rb
@@ -17,7 +17,7 @@ THINGS = [
 ].map(&:freeze).freeze
 
 def verse(n)
-  "This is #{THINGS.last(n).join}.".lines.map(&:chomp)
+  "This is #{THINGS.last(n).join}.".lines.join.tr("\n", ' ')
 end
 
 module Intersperse refine Enumerable do
@@ -29,7 +29,7 @@ end end
 using Intersperse
 
 def verses(a, b)
-  (a..b).map { |n| verse(n) }.intersperse('').flatten(1)
+  (a..b).map { |n| verse(n) }.flatten(1)
 end
 
 json = JSON.parse(File.read(File.join(__dir__, 'canonical-data.json')))
```

This comment only confirms that https://github.com/exercism/problem-specifications/pull/1201/commits/fffe9f5dafe00d4a0b6693ae0831d6e981538020 is consistent. I make no judgment (please do not ask me to) on whether the change is desirable.

**Comment by petertseng** [comment]
Person who merges: Please do remember to squash. Thank you.

**Comment by ErikSchierboom** [comment]
Squashed and merged. Thanks @mRcfps!

**Pull request merged**